tests/core/pyspec/eth2spec/test/phase0/epoch_processing/test_process_registry_updates.py

```diff
@@ -205,16 +214,30 @@ def run_test_ejection_past_churn_limit(spec, state):
 
     yield from run_process_registry_updates(spec, state)
 
+    if is_post_eip7251(spec):
+        per_epoch_churn = spec.get_activation_exit_churn_limit(state)
+
+        def map_index_to_exit_epoch(i):
+            balance_so_far = i * spec.config.EJECTION_BALANCE
+            offset_epoch = balance_so_far // per_epoch_churn
+            if spec.config.EJECTION_BALANCE > per_epoch_churn - (balance_so_far % per_epoch_churn):
+                offset_epoch += 1
+            return expected_ejection_epoch + offset_epoch
```

Comment:
I want to check the math here.. it works but im not sure this is precisely what we want

cc @fradamt

Reply:
I think this currently breaks in the case when `spec.config.EJECTION_BALANCE == per_epoch_churn - (balance_so_far % per_epoch_churn)` and `offset_epoch > 0`, i.e., when the current exit exactly fills up the churn in an epoch after `expected_ejection_epoch`. Here, you wouldn't increase the `offset_epoch`, which makes sense in principle, but doesn't fit the current behavior, where the exit epoch actually jumps to the next one in that case. 
It should be correct with the change proposed here: https://github.com/ethereum/consensus-specs/pull/3668/files#r1562268491

Reply:
Thought it was not going to work in an edge case because of a somewhat strange behavior of the current `compute_exit_epoch_and_update_churn` function, but on a deeper look I think it's always fine as long `spec.config.EJECTION_BALANCE` is not a multiple of the churn limit (which will always be the case unless we increase the ejection balance by a lot)